shared/libraries/config_protocol/src/config_protocol_streaming_consumer.cpp

```diff
@@ -0,0 +1,213 @@
+{
+    const auto coreEventId = static_cast<CoreEventId>(eventArgs.getEventId());
+
+    if (component.assigned())
+    {
+        const auto isSignalOfConnectedClient = [this](const StringPtr& signalId)
+        {
+            std::scoped_lock lock(sync);
+            const auto it = mirroredExternalSignalsIds.find(signalId);
+            return it != mirroredExternalSignalsIds.end();
+        };
+
+        if (externalSignalsFolder.assigned())
+        {
+            if (component == externalSignalsFolder)
+            {
+                if (coreEventId == CoreEventId::ComponentAdded)
+                {
+                    StringPtr addedSignalId = eventArgs.getParameters().get("Component").asPtr<IComponent>().getLocalId();
+                    return !isSignalOfConnectedClient(addedSignalId);
+                }
+                else if (coreEventId == CoreEventId::ComponentRemoved)
+                {
+                    StringPtr removedSignalId = eventArgs.getParameters().get("Id");
+                    return !isSignalOfConnectedClient(removedSignalId);
+                }
+                else if (component.getParent() == externalSignalsFolder)
+                {
+                    return !isSignalOfConnectedClient(component.getLocalId());
+                }
+            }
+        }
+        else if (auto signal = component.asPtrOrNull<ISignal>(); signal.assigned() && isExternalSignal(signal))
+        {
+            return !isSignalOfConnectedClient(component.getLocalId());
+        }
+
+        if (coreEventId == CoreEventId::SignalConnected)
+        {
+            const SignalPtr connectedSignal = eventArgs.getParameters().get("Signal");
+            if (connectedSignal.supportsInterface<IMirroredExternalSignalPrivate>())
+                return !isSignalOfConnectedClient(connectedSignal.getLocalId());
+        }
+    }
+
+    return true;
+}
+
+void ConfigProtocolStreamingConsumer::removeExternalSignals(const ParamsDictPtr& params)
+{
+    ListPtr<IInteger> signalNumericIdsList = params.get("SignalNumericIds");
+    for (const auto& listItem : signalNumericIdsList)
+    {
+        SignalNumericIdType signalNumericId = static_cast<SignalNumericIdType>(listItem);
+        if (const auto it = mirroredExternalSignals.find(signalNumericId); it != mirroredExternalSignals.end())
+        {
+            auto signal = it->second;
+            removeExternalSignal(signal, signalNumericId);
+        }
+    }
+}
+
+void ConfigProtocolStreamingConsumer::addExternalSignal(const MirroredSignalConfigPtr& signal, SignalNumericIdType signalNumericId)
+{
+    {
+        std::scoped_lock lock(sync);
+
+        mirroredExternalSignalsIds.insert(signal.getLocalId());
+        mirroredExternalSignals.insert({signalNumericId, signal});
+    }
+
+    if (externalSignalsFolder.assigned())
+        externalSignalsFolder.addItem(signal);
+
+    LOG_I("Added new mirrored external signal numeric ID {}, remote ID {}, global ID {}",
```

Comment:
Should we log these as info or debug? We want to maybe avoid flooding the log.

Reply:
Changed the log level to debug for all messages from c-2-d streaming components that were previously set to info level.